### PR TITLE
THU/table_edition

### PR DESCRIPTION
**Pull request by csm-thu**
This branch adds several modules with features requested by the Table component in ui and webapp repositories:

- date manipulation functions (parsing, formatting & validation)
- file manipulation functions (read & write CSV, operations on file blob data)
- validation functions (for all variable types supported in the Table)

### CODE REVIEW

**Review by jreynard-code — APPROVED** [review]
LGTM ! ;)

File: package.json [hunk 1]
```diff
@@ -51,8 +51,12 @@
     "typescript": "^4.4.4"
   },
   "dependencies": {
+    "csv-string": "^4.0.1",
+    "date-fns": "^2.25.0",
     "eslint-config-prettier": "^8.3.0",
     "eslint-plugin-prettier": "^4.0.0",
-    "prettier": "^2.4.1"
+    "js-file-download": "^0.4.12",
```
Comment (nborde-CSM) [inline package.json:58]: Is js-file-download still usefull in core lib ?

**Review by nborde-CSM — COMMENTED** [review]
One remark but LGTM

File: package.json (hunk 1 above)
Comment (csm-thu) [reply package.json:58]: Yes, I just added it to provide solutions with the function `downloadFileFromData`

**Pull request merged**